tests/h/paginator_test.py

```diff
@@ -1,158 +1,199 @@
-])
-def test_paginate_returns_offsets_and_limits_resultset(pyramid_request, total, param, offset):
-    pyramid_request.params = {'page': param}
-    query, view = fake_view(total)
-
-    wrapped = paginate_query(view)
-    result = wrapped(context=None, request=pyramid_request)
-
-    assert result['results'].offset_param == offset
-    assert result['results'].limit_param == 20
-
-
-@pytest.mark.parametrize('total,param,meta', [
-    # When there are no items:
-    (0, '1', (1, 1, None, None)),
-    (0, '2', (1, 1, None, None)),
-
-    # When there are N * PAGE_SIZE items:
-    (20, '1', (1, 1, None, None)),
-    (20, '2', (1, 1, None, None)),
-    (40, '1', (1, 2, 2, None)),
-    (40, '2', (2, 2, None, 1)),
-    (40, '3', (2, 2, None, 1)),
-
-    # When the number of items doesn't fit perfectly into N pages:
-    (25, '1', (1, 2, 2, None)),
-    (25, '2', (2, 2, None, 1)),
-    (25, '3', (2, 2, None, 1)),
-
-    # Some extreme cases:
-    (400, '19', (19, 20, 20, 18)),
-
-    # Junk data should be discarded:
-    (100, '999e9', (1, 5, 2, None)),
-    (100, 'fish',  (1, 5, 2, None)),
-    (400, '-19000', (1, 20, 2, None)),
-    (400, '?????', (1, 20, 2, None)),
-])
-def test_paginate_returns_page_metadata(pyramid_request, total, param, meta):
-    pyramid_request.params = {'page': param}
-    query, view = fake_view(total)
-    ecur, emax, enext, eprev = meta
-
-    wrapped = paginate_query(view)
-    result = wrapped(context=None, request=pyramid_request)
-
-    assert result['page'] == {
-        'cur': ecur,
-        'max': emax,
-        'next': enext,
-        'prev': eprev,
-    }
-
-
-def test_paginate_params(pyramid_request):
-    pyramid_request.params = {'page': 2}
-    query, view = fake_view(20)
-
-    decorator = paginate_query(page_size=5)
-    wrapped = decorator(view)
-    result = wrapped(context=None, request=pyramid_request)
-
-    assert result['results'].offset_param == 5
-    assert result['results'].limit_param == 5
+    @pytest.mark.parametrize('page_param,expected', [
+        # Normally 'next' is simply the current page + 1.
+        ('32', 33),
+
+        # If the current page is the last page then 'next' is None.
+        ('60', None),
+    ])
+    def test_next(self, pyramid_request, page_param, expected):
+        pyramid_request.params = {'page': page_param}
+        assert paginate(pyramid_request, 600, 10)['next'] == expected
+
+    @pytest.mark.parametrize('page_param,expected', [
+        # Normally 'prev' is simply the current page - 1.
+        ('32', 31),
+
+        # If the current page is the first page then 'prev' is None.
+        ('1', None),
+    ])
+    def test_prev(self, pyramid_request, page_param, expected):
+        pyramid_request.params = {'page': page_param}
+        assert paginate(pyramid_request, 600, 10)['prev'] == expected
+
+
+@pytest.mark.usefixtures('paginate')
+class TestPaginateQuery(object):
+
+    CURRENT_PAGE = 3
+    "The current page that will be returned by paginate()."
```

Comment:
If you want to document attrs in tests the same way as in the rest of the code then use doc comments:

```python
#: An attr that does stuff
stuff_doer = ...
```